Add Discounts app action link extension template (#420)
Shopify/extensions-templates: discount-app-action-link/README.md

### Get your function ID

`YOUR_DISCOUNT_FUNCTION_ID` (in `intent-schema.json`, `intent-schema-edit.json`, and `instructions.md`) is the ID of the discount function this extension targets. Two ways to find it:

- **CLI** — after `shopify app deploy` (or `shopify app function deploy`), the CLI prints the registered function's ID.
- **Admin API** — query [`shopifyFunctions`](https://shopify.dev/docs/api/admin-graphql/latest/queries/shopifyFunctions):
  ```graphql
  {
    shopifyFunctions(first: 50) {
      nodes {
        id
        title
        description
      }
    }
  }
  ```

### Disambiguation (multiple functions)

If your app ships more than one discount function, register one create intent per function (each in its own extension) and use the `functionId` `matchValue` to narrow each intent to a single function. Each registered intent counts toward the per-app intent limit below.

### Limits

See the [Sidekick app action limits](https://shopify.dev/docs/apps/build/sidekick#limits) in the developer docs.

### Testing

Run `shopify app deploy` to deploy your extensions and test them with Sidekick in the Shopify Admin.

> > **Note:** `shopify app dev` may not upload the intent schema correctly. Use `shopify app deploy` for testing.

Reviewer: This is out of date too. We definitely should use `shopify app dev` only

Apps that power discounts with a function typically want both intents so Sidekick can drive the full create + edit flow. If your app only supports one, delete the other `[[extensions]]` block in `shopify.extension.toml`.

Learn more about app action extensions in Shopify's [developer documentation](https://shopify.dev/docs/apps/build/sidekick/build-app-actions).

---

## Get started with this extension

The module defines two `admin.app.intent.link` extensions, each registering one intent:

- **Create** (`action = "create"`) — opens your app's discount creation flow. Pinned to a single discount function via the `functionId` `matchValue` in `intent-schema.json`.
- **Edit** (`action = "edit"`) — opens your app's discount edit flow for a specific discount. Maps the discount GID to the `:id` route param.

Both intents are navigation-only: Sidekick routes the merchant to your app to complete the action, and the intents alone make the extension discoverable by Sidekick (no `tools.json` needed). The create intent ships an `instructions.md` telling Sidekick to pass your `functionId` so it resolves directly to your app's create flow.

> Each `admin.app.intent.link` extension registers exactly one intent, which is why create and edit are two `[[extensions]]` blocks (with distinct handles `{handle}-create` and `{handle}-edit`) rather than one.

Reviewer: This is not true, you can have multiple intents. This limitation is on the url
```suggestion
Each `admin.app.intent.link` extension is tied to only a single url, which is why create and edit are two `[[extensions]]` blocks (with distinct handles `{handle}-create` and `{handle}-edit`) rather than one.
```

Author: Done!

3. Update the merchant-facing names and descriptions in `locales/en.default.json` (under the `create` and `edit` keys).

### Get your function ID

`YOUR_DISCOUNT_FUNCTION_ID` (in `intent-schema.json` and `instructions.md`) is the ID of the discount function this extension targets. Two ways to find it:

- **CLI** — after `shopify app deploy` (or `shopify app function deploy`), the CLI prints the registered function's ID.
- **Admin API** — query [`shopifyFunctions`](https://shopify.dev/docs/api/admin-graphql/latest/queries/shopifyFunctions):
  ```graphql
  {
    shopifyFunctions(first: 50) {
      nodes {
        id
        title
        description
      }
    }
  }
  ```

### Disambiguation (multiple functions)

If your app ships more than one discount function, register one create intent per function (each in its own extension) and use the `functionId` `matchValue` to narrow each intent to a single function. Each registered intent counts toward the per-app intent limit below.

> ### Limits

Reviewer: Instead of listing the limits here which could get out to sync can we link to the [docs](https://shopify.dev/docs/apps/build/sidekick#limits)?

Author: Done!
